caravel/templates/index.html

```diff
@@ -5,8 +5,40 @@
   <script src="/static/js/continuation.js" type="text/javascript"></script>
 {% endblock %}
 {% block content %}
-  {% for listing in listings %}
-    {% include "listings/inline.html" %}
-  {% endfor %}
-  </div>
+    <button type="button" class="btn btn-default"><a href="{{ url_for('new_listing') }}" style="text-decoration: none; color: black;">New Listing</a></button>
+    {% if view == "Thumbnail" %}
+      <form method="get" style="display:inline-block; float:right">
+      <input class="btn btn-default" type="submit" name="v" value="List">
```

Comment:
something something `onchange="this.parentNode.submit()"` something something probably

More the idea is of changing as soon as possible, rather than waiting for a button click.